packages/mcp-server/src/lib/transports/http-transport.ts

```
@@ -79,18 +82,82 @@
 
 		// Cookie parsing for session management
 		this.app.use(cookieParser());
 
 		// URL-encoded body parsing (e.g., form submissions)
 		this.app.use(express.urlencoded({ extended: true }));
 
+		// Error handling middleware for request validation errors
+		this.app.use((err: any, req: express.Request, res: express.Response, next: express.NextFunction): void => {
+			// Skip if response already sent
+			if (res.headersSent) {
+				return next(err);
```

Biome (2.1.2)

[error] The function should not return a value because its return type is void.

The function is here:

'void' signals the absence of value. The returned value is likely to be ignored by the caller.

(lint/correctness/noVoidTypeReturn)